### DISCUSSION TITLE
how to get week info of a date/datetime filed?

### DISCUSSION THREAD

**Discussion by flyly0755** [sqlalchemy/sqlalchemy#8067] [Usage Questions] [answered]
now i know how to get year, month, day of a date/datetime filed with sqlalchemy showed as below: 
from sqlalchemy import func as F
F.extract('year', datefield).label('YEAR'),
F.extract('month', datefield).label('MONTH'),
F.extract('day', datefield).label('DAY')

but how can i get which week  of a date/datetime filed in a year, seems a little complicated：
https://clickhouse.com/docs/en/sql-reference/functions/date-time-functions/ with this link about toWeek() method, we can see week info has several mode:
Mode | First day of week | Range | Week 1 is the first week …
-- | -- | -- | --
0 | Sunday | 0-53 | with a Sunday in this year
1 | Monday | 0-53 | with 4 or more days this year
2 | Sunday | 1-53 | with a Sunday in this year
3 | Monday | 1-53 | with 4 or more days this year
4 | Sunday | 0-53 | with 4 or more days this year
5 | Monday | 0-53 | with a Monday in this year
6 | Sunday | 1-53 | with 4 or more days this year
7 | Monday | 1-53 | with a Monday in this year
8 | Sunday | 1-53 | contains January 1
9 | Monday | 1-53 | contains January 1
how can i get similar result with sqlalchemy package?

**Comment by zzzeek** [comment, marked as answer]
these are standard-syntax SQL functions which you render using the [func](https://docs.sqlalchemy.org/en/14/core/sqlelement.html?highlight=func#sqlalchemy.sql.expression.func) namespace.  A tutorial on using SQL functions is at https://docs.sqlalchemy.org/en/14/tutorial/data_select.html#working-with-sql-functions .   Concretely, given the link you gave:


```py
"""
SELECT
    toDateTime('2016-06-15 23:00:00') AS time,
    toDate(time) AS date_local,
    toDate(time, 'Asia/Yekaterinburg') AS date_yekat,
    toString(time, 'US/Samoa') AS time_samoa
"""

from sqlalchemy import func, select

time = func.toDateTime('2016-06-15 23:00:00').label("time")
stmt = select(
    time,
    func.toDate(time).label("date_local"),
    func.toDate(time, 'Asia/Yekaterinburg').label("date_yekat"),
    func.toString(time, 'US/Samoa').label("time_samoa")
)

row = connection.execute(stmt).first()

```

**Reply by zzzeek** [reply]
e.g.   `func.toWeek(date, 5)`

**Reply by flyly0755** [reply]
thanks, this is what i want:)
